### PR TITLE
[FIX] Add Goblin Shovel Stealer with Rusty Shovel

### PR DESCRIPTION
**Pull request by LittleEins**
# Description

Stemmed from https://discord.com/channels/880987707214544966/950040644754235452/1000918317768376362:

- Create jumping goblin shovel stealer holding rusty shovel.
- Add logic that shows a goblin shovel stealer holding rusty shovel when Shovel is not crafted yet.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
goblin_jump_rusty_shovel

# Goblin Stealer Holding Rusty Shovel
This will show when Shovel is not crafted yet.
proof
# Goblin Stealer Holding Shovel
This will show when Shovel is crafted.
image

# How Has This Been Tested?
Ran on local testnet with screenshots above.

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by veenhdek** [comment]
Also did all of this process, lol good work @LittleEins beat me to the PR. goblin_jump_rusty_shovel

Will add the syntax clarifications i had put in mine in the comments

File: src/features/farming/crops/components/GoblinShovel.tsx [hunk 1]
```diff
@@ -87,6 +88,9 @@ export const GoblinShovel: React.FC = () => {
   };
 
   const shovelImage = state.inventory.Shovel?.gte(1) ? shovel : rustyShovel;
+  const goblinstealerState = state.inventory.Shovel?.gte(1)
```
Comment (veenhdek) [inline src/features/farming/crops/components/GoblinShovel.tsx:91]: Suggestion: leave this just as goblin to not get too wordy and add a comment:
`//if the normal shovel hasn't been crafted show rusty instead of normal shovel`
  `const shovelImage = state.inventory.Shovel?.gte(1) ? shovel : rustyShovel;`
  `const goblin = state.inventory.Shovel?.gte(1) ? shovelGoblin : rustyShovelGoblin;`

File: src/features/farming/crops/components/GoblinShovel.tsx [hunk 2]
```diff
@@ -116,7 +120,7 @@ export const GoblinShovel: React.FC = () => {
 
       {showGoblin && (
         <img
-          src={goblin}
+          src={goblinstealerState}
```
Comment (veenhdek) [inline src/features/farming/crops/components/GoblinShovel.tsx:123]: Suggestion: leave as -  `src={goblin}` if all other suggestions taken

File: src/features/farming/crops/components/GoblinShovel.tsx [hunk 3]
```diff
@@ -4,6 +4,7 @@ import React, { useContext, useEffect, useState } from "react";
 import { useActor } from "@xstate/react";
 import Modal from "react-bootstrap/Modal";
 import goblin from "assets/npcs/goblin_jump_shovel.gif";
+import rustyGoblin from "assets/npcs/goblin_jump_rusty_shovel.gif";
```
[the remark this answers is not in the document]
Comment (LittleEins) [reply src/features/farming/crops/components/GoblinShovel.tsx]: Thank you for the suggestion <3 I already commited new changes based on it

**Comment by veenhdek** [comment]
Love it @LittleEins

**Review by Rofrtd — APPROVED** [review]
Nice one! 
LGTM! 🚀
